sodasql/scan/dialect.py

```diff
@@ -365,3 +367,13 @@ def is_connection_error(self, exception):
 
     def is_authentication_error(self, exception):
         return False
+
+    def try_to_raise_soda_sql_exception(self, exception: Exception) -> Exception:
+        if self.is_connection_error(exception):
+            raise WarehouseConnectionError(error_code=ERROR_CODE_CONNECTION_FAILED,
```

Comment:
I would recommend moving the error code IDs into the constructor since they are anyway linked with the exception class, so this calls ends up without `error_code=` argument

Reply:
Indeed, I also thought this was not very useful.